crypto/src/main/java/org/springframework/security/crypto/bcrypt/BCryptPasswordEncoder.java

```diff
@@ -119,6 +119,11 @@ public boolean matches(CharSequence rawPassword, String encodedPassword) {
 			return false;
 		}
 
+		if(rawPassword.toString() == null || rawPassword.toString.length() == 0){
+			logger.warn("Empty raw password");
+			return false;
+		}
+
```

Comment:
Would you also add a couple of tests to `BCryptPasswordEncoderTests`? I think that `doesntMatchNullEncodedValue` and `doesntMatchEmptyEncodedValue` are probably good examples to follow.